--- linux-old/net/ipv4/netfilter/ipt_ah.c	Fri Mar 22 10:57:15 2002
+++ linux/net/ipv4/netfilter/ipt_ah.c	Fri Jul  5 16:05:06 2002
@@ -91,12 +91,12 @@
 static struct ipt_match ah_match
 = { { NULL, NULL }, "ah", &match, &checkentry, NULL, THIS_MODULE };
 
-int __init init(void)
+static int __init init(void)
 {
 	return ipt_register_match(&ah_match);
 }
 
-void __exit cleanup(void)
+static void __exit cleanup(void)
 {
 	ipt_unregister_match(&ah_match);
 }
